lib/jxl/encode.cc

```diff
@@ -323,6 +323,9 @@ JxlEncoderStatus JxlEncoderOptionsSetInteger(JxlEncoderOptions* options,
       options->values.cparams.resampling = value;
       return JXL_ENC_SUCCESS;
     case JXL_ENC_OPTION_EXTRA_CHANNEL_RESAMPLING:
+      // TOOD(lode): the jxl codestream allows choosing a different resampling
+      // factor for each extra channel, move this to a per-extra-channel setting
```

Comment:
It's per frame and per extra channel, so cannot be moved to extrachannelinfo (which is global for the whole image).
We should also check that the total upsampling factor of extra channels (ec_resampling << dim_shift) is at least as large as the upsampling factor of the color image, since that is a spec requirement and the encoder and decoder will choke if that constraint is not satisfied.

Reply:
Updated the TODO to make clear it's per frame and per extra channel and how to implement it